src/middleware.ts

```diff
@@ -0,0 +1,29 @@
+import { NextRequest, NextResponse } from 'next/server'
+
+import { DEFAULT_LANGUAGE } from './constants';
+import { stringifyUrlObject } from './utils/routerUtils';
+
+const PUBLIC_FILE = /\.(.*)$/
+/**
+ * Enforce prefix for default locale 'fi'
+ * https://github.com/vercel/next.js/discussions/18419
+ * @param req
+ */
+
+export async function middleware(req: NextRequest) {
```

Comment:
This middleware should be named (and packaged?) more specifically. For example "localeRouteMiddleware".

Reply:
The latest stable version of middleware feature has the decision that there cannot be nested middlwares, but only one middleware instead: https://nextjs.org/docs/messages/middleware-upgrade-guide#no-nested-middleware
So renaming would probably just confuse that default logic. Of course it means that this single middleware can have multiple purposes, so perhaps it could have different methods called, depending on purpose.

Reply:
now cleaned a little bit

Reply:
I don't like the changes that the NExt.JS team has made 😆